### PR TITLE
Added validation check for resource group name and other fields in Azure VM Agent Cloud

### PR DESCRIPTION
**Pull request by arroyc**
@mmitche  @clguimanMSFT

### CODE REVIEW

File: src/main/java/com/microsoft/azure/AzureVMCloud.java [hunk 1]
```diff
@@ -690,16 +690,24 @@ public String getDefaultResourceGroupName() {
         }
 
         public FormValidation doVerifyConfiguration(
+                @QueryParameter String subscriptionId,
```
Comment (arroyc) [inline src/main/java/com/microsoft/azure/AzureVMCloud.java]: removed

File: src/main/java/com/microsoft/azure/AzureVMCloud.java [hunk 2]
```diff
@@ -690,16 +690,24 @@ public String getDefaultResourceGroupName() {
...
+                @QueryParameter String deploymentTimeout,
                 @QueryParameter String azureCredentialsId,
                 @QueryParameter String resourceGroupName) {
 
             if (StringUtils.isBlank(resourceGroupName)) {
                 resourceGroupName = Constants.DEFAULT_RESOURCE_GROUP_NAME;
             }
 
+            
             AzureCredentials.ServicePrincipal credentials = AzureCredentials.getServicePrincipal(azureCredentialsId);
```
Comment (arroyc) [inline src/main/java/com/microsoft/azure/AzureVMCloud.java]: didn't get you !!

Comment (clguimanMSFT) [inline src/main/java/com/microsoft/azure/AzureVMCloud.java]: azureCredentialsId was missing or I missed seeing it as a method argument

File: src/main/java/com/microsoft/azure/AzureVMManagementServiceDelegate.java [hunk 3]
```diff
@@ -632,14 +638,16 @@ public static String verifyConfiguration(
             return Messages.Azure_GC_Template_Val_Profile_Missing();
         } else {
             try {
+                if (!resourceGroupName.matches(Constants.DEFAULT_RESOURCE_GROUP_PATTERN)) {
```
Comment (arroyc) [inline src/main/java/com/microsoft/azure/AzureVMManagementServiceDelegate.java]: serviceprincipal.validate  checkes arg, vm limit and timeout, here we only check rg. If we need to check all 3 then we need to change verify configuration as well. should I do it?

**Pull request merged**